### PR TITLE
feat(curriculum): create CSS Pseudo Classes quiz

### LINKED ISSUE #56183: Pseudo Classes

### PR DESCRIPTION
**Pull request by SahilWMI**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine or GitPod.

Closes #56183

### CODE REVIEW

File: curriculum/challenges/english/25-front-end-development/quiz-css-pseudo-classes/66ed9002f45ce3ece4053eb6.md [hunk 1]
```diff
@@ -17,439 +17,509 @@ Answer all of the questions below correctly to pass the quiz.
...
 
-Placeholder distractor 3
+They are selector used to add movement effects to an element during interactions.
 
 #### --answer--
 
-Placeholder answer
+They are keywords added to a selector that change how an element looks or behaves when it's in a specific state
+
+## --quiz--
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/quiz-css-pseudo-classes/66ed9002f45ce3ece4053eb6.md:38]: you should not add new headers, please go through the file and remove this header. there is only one per file at the top

File: curriculum/challenges/english/25-front-end-development/quiz-css-pseudo-classes/66ed9002f45ce3ece4053eb6.md [hunk 2]
```diff
@@ -17,439 +17,509 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 #### --text--
 
-Placeholder question
+Which pseudo-class applies when you hover over an element?
 
 #### --distractors--
 
-Placeholder distractor 1
+:focus
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/quiz-css-pseudo-classes/66ed9002f45ce3ece4053eb6.md:48]:
```suggestion
`:focus`
```

use backticks around code things, please make this change in all the file

File: curriculum/challenges/english/25-front-end-development/quiz-css-pseudo-classes/66ed9002f45ce3ece4053eb6.md [hunk 3]
```diff
@@ -17,439 +17,509 @@ Answer all of the questions below correctly to pass the quiz.
...
 ---
 
-Placeholder distractor 3
+:where()
 
 #### --answer--
 
-Placeholder answer
-
+::before
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/quiz-css-pseudo-classes/66ed9002f45ce3ece4053eb6.md:525]:
```suggestion
`::before`

```

let's not remove EOF lines

**Review by majestic-owl448 — COMMENTED** [review]
please update the title of the PR to follow the title guidelines

**Review by SahilWMI — COMMENTED** [review]
sorry for the mistakes, can you review and tell me if it needs any other correction?

corrections made so far:

- removeed quiz placeholdeer

- added backticks around lines of code.

File: curriculum/challenges/english/25-front-end-development/quiz-css-pseudo-classes/66ed9002f45ce3ece4053eb6.md [hunk 4]
```diff
@@ -17,439 +17,474 @@ Answer all of the questions below correctly to pass the quiz.
...
 ---
 
-Placeholder distractor 3
+`:checked`
 
 #### --answer--
 
-Placeholder answer
+`:active`
+
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/quiz-css-pseudo-classes/66ed9002f45ce3ece4053eb6.md:104]:
```suggestion
`:active`

```

let's not add extra empty lines

File: curriculum/challenges/english/25-front-end-development/quiz-css-pseudo-classes/66ed9002f45ce3ece4053eb6.md [hunk 5]
```diff
@@ -17,439 +17,474 @@ Answer all of the questions below correctly to pass the quiz.
...
+  color: red;
+}
+```
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+_____ pseudo-class is used for the current active link.
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/quiz-css-pseudo-classes/66ed9002f45ce3ece4053eb6.md:339]:
```suggestion
Which pseudo-class is used for the current active link?
```

File: curriculum/challenges/english/25-front-end-development/quiz-css-pseudo-classes/66ed9002f45ce3ece4053eb6.md [hunk 6]
```diff
@@ -17,439 +17,474 @@ Answer all of the questions below correctly to pass the quiz.
...
 ---
 
-Placeholder distractor 3
+It selects elements that match a given selector.
 
 #### --answer--
 
-Placeholder answer
+It selects elements that do not match a given selector.
+
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/quiz-css-pseudo-classes/66ed9002f45ce3ece4053eb6.md:378]:
```suggestion
It selects elements that do not match a given selector.
```

an other extra line

File: curriculum/challenges/english/25-front-end-development/quiz-css-pseudo-classes/66ed9002f45ce3ece4053eb6.md [hunk 7]
```diff
@@ -17,439 +17,470 @@ Answer all of the questions below correctly to pass the quiz.
 
 #### --text--
 
-Placeholder question
+What are pseudo-classes?
 
 #### --distractors--
 
-Placeholder distractor 1
+It is a CSS rule that creates additional content within an element when needed.
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-css-pseudo-classes/66ed9002f45ce3ece4053eb6.md:24]:
```suggestion
They are CSS rules that create additional content within an element when needed.
```

File: curriculum/challenges/english/25-front-end-development/quiz-css-pseudo-classes/66ed9002f45ce3ece4053eb6.md [hunk 8]
```diff
@@ -17,439 +17,470 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 ---
 
-Placeholder distractor 2
+It is a property that adjusts an element's size or layout on the page.
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-css-pseudo-classes/66ed9002f45ce3ece4053eb6.md:28]:
```suggestion
They are properties that adjust an element's size or layout on the page.
```

File: curriculum/challenges/english/25-front-end-development/quiz-css-pseudo-classes/66ed9002f45ce3ece4053eb6.md [hunk 9]
```diff
@@ -17,439 +17,470 @@ Answer all of the questions below correctly to pass the quiz.
...
 
-Placeholder answer
+`:active`
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Which pseudo-class is used to select and style when an element is focused?
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-css-pseudo-classes/66ed9002f45ce3ece4053eb6.md:108]:
```suggestion
Which pseudo-class is used to select and style an element when it's focused?
```

File: curriculum/challenges/english/25-front-end-development/quiz-css-pseudo-classes/66ed9002f45ce3ece4053eb6.md [hunk 10]
```diff
@@ -17,439 +17,470 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 #### --distractors--
 
-Placeholder distractor 1
+It styles your element when you click on it.
 
 ---
 
-Placeholder distractor 2
+It makes your element float around the screen
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-css-pseudo-classes/66ed9002f45ce3ece4053eb6.md:138]:
```suggestion
It makes your element float around the screen.
```

File: curriculum/challenges/english/25-front-end-development/quiz-css-pseudo-classes/66ed9002f45ce3ece4053eb6.md [hunk 11]
```diff
@@ -17,439 +17,470 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 ---
 
-Placeholder distractor 3
+It hides elements on click
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-css-pseudo-classes/66ed9002f45ce3ece4053eb6.md:142]:
```suggestion
It hides elements on click.
```

File: curriculum/challenges/english/25-front-end-development/quiz-css-pseudo-classes/66ed9002f45ce3ece4053eb6.md [hunk 12]
```diff
@@ -17,439 +17,470 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 #### --answer--
 
-Placeholder answer
+It styles an element when you hover over it
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-css-pseudo-classes/66ed9002f45ce3ece4053eb6.md:146]:
```suggestion
It styles an element when you hover over it.
```

File: curriculum/challenges/english/25-front-end-development/quiz-css-pseudo-classes/66ed9002f45ce3ece4053eb6.md [hunk 13]
```diff
@@ -17,439 +17,470 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 #### --text--
 
-Placeholder question
+What does the `:disabled` pseudo-class do?
 
 #### --distractors--
 
-Placeholder distractor 1
+It styles checked inputs
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-css-pseudo-classes/66ed9002f45ce3ece4053eb6.md:238]:
```suggestion
It styles checked inputs.
```

File: curriculum/challenges/english/25-front-end-development/quiz-css-pseudo-classes/66ed9002f45ce3ece4053eb6.md [hunk 14]
```diff
@@ -17,439 +17,470 @@ Answer all of the questions below correctly to pass the quiz.
...
 
-Placeholder answer
+`:valid`
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Which one of these is not a Location pseudo-class?
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-css-pseudo-classes/66ed9002f45ce3ece4053eb6.md:278]:
```suggestion
Which one of these is not a location pseudo-class?
```

File: curriculum/challenges/english/25-front-end-development/quiz-css-pseudo-classes/66ed9002f45ce3ece4053eb6.md [hunk 15]
```diff
@@ -17,439 +17,470 @@ Answer all of the questions below correctly to pass the quiz.
...
 
-Placeholder answer
+`:current`
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Which of the following selects the 3rd child in a row?
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-css-pseudo-classes/66ed9002f45ce3ece4053eb6.md:300]:
```suggestion
Which of the following selects the third list item?
```

File: curriculum/challenges/english/25-front-end-development/quiz-css-pseudo-classes/66ed9002f45ce3ece4053eb6.md [hunk 16]
```diff
@@ -17,439 +17,470 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 ---
 
-Placeholder distractor 3
+It styles odd children
 
 #### --answer--
 
-Placeholder answer
+It selects the 4th child
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-css-pseudo-classes/66ed9002f45ce3ece4053eb6.md:398]:
```suggestion
It selects the fourth child.
```

File: curriculum/challenges/english/25-front-end-development/quiz-css-pseudo-classes/66ed9002f45ce3ece4053eb6.md [hunk 17]
```diff
@@ -17,439 +17,470 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 #### --text--
 
-Placeholder question
+What does the `:nth-child(4)` pseudo-class do?
 
 #### --distractors--
 
-Placeholder distractor 1
+It selects the last child
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-css-pseudo-classes/66ed9002f45ce3ece4053eb6.md:386]:
```suggestion
It selects the last child.
```

File: curriculum/challenges/english/25-front-end-development/quiz-css-pseudo-classes/66ed9002f45ce3ece4053eb6.md [hunk 18]
```diff
@@ -17,439 +17,470 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 ---
 
-Placeholder distractor 2
+It selects all children
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-css-pseudo-classes/66ed9002f45ce3ece4053eb6.md:390]:
```suggestion
It selects all children.
```

File: curriculum/challenges/english/25-front-end-development/quiz-css-pseudo-classes/66ed9002f45ce3ece4053eb6.md [hunk 19]
```diff
@@ -17,439 +17,470 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 ---
 
-Placeholder distractor 3
+It styles odd children
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-css-pseudo-classes/66ed9002f45ce3ece4053eb6.md:394]:
```suggestion
It styles odd children.
```

File: curriculum/challenges/english/25-front-end-development/quiz-css-pseudo-classes/66ed9002f45ce3ece4053eb6.md [hunk 20]
```diff
@@ -17,439 +17,470 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 #### --text--
 
-Placeholder question
+What does the `:last-of-type` pseudo class do?
 
 #### --distractors--
 
-Placeholder distractor 1
+It selects the first child element
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-css-pseudo-classes/66ed9002f45ce3ece4053eb6.md:408]:
```suggestion
It selects the first child element.
```

File: curriculum/challenges/english/25-front-end-development/quiz-css-pseudo-classes/66ed9002f45ce3ece4053eb6.md [hunk 21]
```diff
@@ -17,439 +17,470 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 ---
 
-Placeholder distractor 2
+It selects the middle element
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-css-pseudo-classes/66ed9002f45ce3ece4053eb6.md:412]:
```suggestion
It selects the middle element.
```

File: curriculum/challenges/english/25-front-end-development/quiz-css-pseudo-classes/66ed9002f45ce3ece4053eb6.md [hunk 22]
```diff
@@ -17,439 +17,470 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 ---
 
-Placeholder distractor 3
+It selects every element in a group
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-css-pseudo-classes/66ed9002f45ce3ece4053eb6.md:416]:
```suggestion
It selects every element in a group.
```

File: curriculum/challenges/english/25-front-end-development/quiz-css-pseudo-classes/66ed9002f45ce3ece4053eb6.md [hunk 23]
```diff
@@ -17,439 +17,470 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 #### --answer--
 
-Placeholder answer
+It selects the last child element
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-css-pseudo-classes/66ed9002f45ce3ece4053eb6.md:420]:
```suggestion
It selects the last sibling element.
```

File: curriculum/challenges/english/25-front-end-development/quiz-css-pseudo-classes/66ed9002f45ce3ece4053eb6.md [hunk 24]
```diff
@@ -17,439 +17,474 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Use ____ to select the second item in a list.
```
[the remark this answers is not in the document]
Comment (naomi-lgbt) [reply curriculum/challenges/english/25-front-end-development/quiz-css-pseudo-classes/66ed9002f45ce3ece4053eb6.md:426]: This one has not been addressed.

File: curriculum/challenges/english/25-front-end-development/quiz-css-pseudo-classes/66ed9002f45ce3ece4053eb6.md [hunk 25]
```diff
@@ -17,439 +17,474 @@ Answer all of the questions below correctly to pass the quiz.
...
 ---
 
-Placeholder distractor 3
+`:where()`
 
 #### --answer--
 
-Placeholder answer
-
+`::before`
```
[the remark this answers is not in the document]
Comment (naomi-lgbt) [reply curriculum/challenges/english/25-front-end-development/quiz-css-pseudo-classes/66ed9002f45ce3ece4053eb6.md:486]: This one is not resolved either.

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]

**Review by SahilWMI — COMMENTED** [review]
Fixed grammar, terminologies and updated questions.

**Review by SahilWMI — COMMENTED** [review]
fixed even more grammar and added EOF-lines.

**Review by naomi-lgbt — APPROVED** [review]

**Review by majestic-owl448 — APPROVED** [review]

**Comment by ilenia-magoni** [comment]
Hi @SahilWMI . Congrats on your first pull request (PR)! 🎉

Thank you for your contribution to the page! 👍
We are happy to accept these changes and look forward to future contributions. 📝

**Pull request merged**